starknet_py/cairo/data_types.py

```diff
@@ -59,10 +59,41 @@ class StructType(CairoType):
 
 @dataclass
 class EnumType(CairoType):
+    """
+    Type representation of Cairo enums.
+    """
+
     name: str
     variants: OrderedDict[str, CairoType]
 
 
 @dataclass
 class Option(CairoType):
-    value: Optional[CairoType] = None
+    """
+    Type representation of Cairo options.
+    """
+
+    type: Optional[CairoType] = None
+
+
+@dataclass
+class UintType(CairoType):
+    """
+    Type representation of Cairo unsigned integers.
+    """
+
+    bits: int
+
+    def check_range(self, value: int):
+        """
+        Utility method checking if the `value` is in range.
+        """
```

Comment:
How about distinguishing between different types of cairo enums? I think it might be necessary in case of converting python -> starknet callldata.